packages/dds/tree/src/test/forestTestSuite.ts

```diff
@@ -0,0 +1,724 @@
+                forest.applyDelta(delta);
+
+                const reader = forest.allocateCursor();
+                moveToDetachedField(forest, reader);
+                assert(reader.firstNode());
+                reader.enterField(xField);
+                assert.equal(reader.getFieldLength(), 0);
+                reader.exitField();
+                reader.enterField(yField);
+                assert.equal(reader.getFieldLength(), 2);
+            });
+
+            it("insert and modify", () => {
+                const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+                const content: JsonableTree[] = [
+                    { type: jsonNumber.name, value: 1 },
+                    { type: jsonNumber.name, value: 2 },
+                ];
+                initializeForest(forest, content.map(singleTextCursor));
+
+                const mark: Delta.InsertAndModify = {
+                    type: Delta.MarkType.InsertAndModify,
+                    content: singleTextCursor({ type: jsonNumber.name, value: 3 }),
+                    fields: new Map([
+                        [
+                            brand("newField"),
+                            [
+                                {
+                                    type: Delta.MarkType.Insert,
+                                    content: [{ type: jsonNumber.name, value: 4 }].map(
+                                        singleTextCursor,
+                                    ),
+                                },
+                            ],
+                        ],
+                    ]),
+                };
+                const delta: Delta.Root = new Map([[rootFieldKeySymbol, [mark]]]);
+                // TODO: make type-safe
+                forest.applyDelta(delta);
+
+                const reader = forest.allocateCursor();
+                moveToDetachedField(forest, reader);
+                assert(reader.firstNode());
+                assert.equal(reader.value, 3);
+                reader.enterField(brand("newField"));
+                assert(reader.firstNode());
+                assert.equal(reader.value, 4);
+                assert.equal(reader.nextNode(), false);
+                reader.exitField();
+                assert.equal(reader.nextNode(), true);
+                assert.equal(reader.value, 1);
+                assert.equal(reader.nextNode(), true);
+                assert.equal(reader.value, 2);
+                assert.equal(reader.nextNode(), false);
+            });
+
+            it("modify and delete", () => {
+                const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+                initializeForest(forest, nestedContent.map(singleTextCursor));
+
+                const mark: Delta.ModifyAndDelete = {
+                    type: Delta.MarkType.ModifyAndDelete,
+                    fields: new Map([[xField, []]]),
```

Comment:
To test ModifyAndDelete we need MoveOut under the ModifyAndDelete. Whas this an unsupported case by the current visitor implementation?

Reply:
Ah, I just didn't quite understand the intended usage of this delta. So to be clear, this is so that we can move out some nodes before deleting the parent?

Reply:
That's correct.